### PR TITLE
Streamline GitHub Actions

### PR DESCRIPTION
**Pull request by qbarrand**
Move the build of the manager binary to the unit-tests Prow script.
Remove the linting of the code in favor of the lint Prow script.

### CODE REVIEW

File: ci/prow/lint [hunk 1]
```diff
@@ -2,4 +2,13 @@
 
 set -euxo pipefail
 
+# Generate code and compare with the current source.
+# Fail with a diff if some files are outdated.
+make generate
+
+if ! git diff --exit-code; then
+    echo "Changes detected; diff:"
+    git diff
```
Comment (ybettan) [inline ci/prow/lint:11]: I don't think we need to print `diff` as the `-x` of the script will do it for us.

File: ci/prow/unit-tests [hunk 2]
```diff
@@ -3,3 +3,4 @@
 set -euxo pipefail
 
 make unit-test
+make manager
```
Comment (ybettan) [inline ci/prow/unit-tests:6]: Why is this required?

If it is only to test the the code is compiled correctly, then `unit-test` will fail (because `vet` will fail)

Comment (ybettan) [reply ci/prow/unit-tests:6]: Shouldn't it be part of the `lint` job in that case?

Comment (qbarrand) [reply ci/prow/unit-tests:6]: `lint` runs in a golangci-lint image, not sure it can build our code and with which go version.
Unit tests do not need to build the manager binary to run, so I wouldn't add it as a dependency.
Maybe we should have a separate `build` job then?

Comment (ybettan) [reply ci/prow/unit-tests:6]: > Maybe we should have a separate build job then?

I think so if you don't mind.

**Comment by qbarrand** [comment]
/assign @yevgeny-shnaidman
This PR needs a `/lgtm`. Thanks!
